styx-scheduler-service/src/test/java/com/spotify/styx/SchedulerTest.java

```diff
@@ -150,8 +159,10 @@ public void setUp() throws Exception {
     when(stateManager.getActiveState(any())).then(a ->
         Optional.ofNullable(activeStates.get(a.<WorkflowInstance>getArgument(0))));
 
+    when(random.nextInt(anyInt())).then(a -> a.<Integer>getArgument(0) - 1);
```

Comment:
This is dirty, but otherwise hard to test.

Reply:
I factored out the shuffling to avoid using Random